src/api/catalog.ts

```diff
@@ -58,6 +58,14 @@ export function search({
     default:
       return Promise.reject(new Error(`Unknown data source prefix: '${source}'`))
   }
+  let from = new Date(dateFrom)
```

Comment:
OK cool.  Right now, the pattern will prevent weird things like "foo" to be entered, and this test will prevent things like 9999-99-99 from being entered (or at least will catch it before it hangs)

I forgot momentjs was available so I can add a check that "to" is before "from" and probably even check that the date is valid through moment as well.